### ISSUE TITLE
Make EarlyStopping throw an error on unknown modes

### ISSUE THREAD

**Issue by alanhdu** [PyTorchLightning/pytorch-lightning#5252]
## 🚀 Feature

Right now, passing in a nonsense `mode` to `pl.callbacks.EarlyStopping` passes silently
```python
>>> import pytorch_lightning as pl
>>> pl.callbacks.EarlyStopping("val_f1", mode="werwer")
```

In the code, it looks  this is deliberate: in this case, we only raise a warning when the verbosity is >0
https://github.com/PyTorchLightning/pytorch-lightning/blob/6adc1b32bdeddbc34282abe0fd9f654c0cba570b/pytorch_lightning/callbacks/early_stopping.py#L104-L110

I think this is undesirable behavior -- if you make a typo when passing in a parameter to `mode,` `EarlyStopping` should fail with an exception, not just produce a warning about potentially incorrect behavior.  Worse, `EarlyStopping` only outputs a warning when the `verbosity > 0` -- that means by default, `EarlyStopping` will just *silently* just produce incorrect behavior.

**Comment by alanhdu** [comment]
It looks like `pl.callbacks.ModelCallback` has the same issue, although it seems to unconditionally raise a warning about an unknown mode:
```python
>>> p = pl.callbacks.ModelCheckpoint("val_f1", mode="werwer")
/opt/anaconda/envs/ctrldev/lib/python3.6/site-packages/pytorch_lightning/utilities/distributed.py:49: RuntimeWarning: ModelCheckpoint mode werwer is unknown, fallback to auto mode
  warnings.warn(*args, **kwargs)
```

I believe that this should also thrown an exception, not just a warning.

**Comment by rohitgr7** [comment]
auto mode is deprecated already, but yeah we can remove that verbose check from EarlyStopping in the meantime.

**Comment by alanhdu** [comment]
> auto mode is deprecated already, but yeah we can remove that verbose check from EarlyStopping in the meantime.

Sorry if I wasn't clear -- I think this is unrelated to the `auto` deprecation. I'm saying that if I pass in an "unknown" `mode` (i.e. one that is not `max`, `min`, or `auto`), then IMO `EarlyStopping` and `ModelCheckpoint` should throw an exception, not just log a warning (since that's probably a user error).

**Comment by alanhdu** [comment]
https://github.com/PyTorchLightning/pytorch-lightning/pull/5255 is the behavior I was thinking of -- given the size of the change, I thought I'd just put together the PR, but happy to change it based on the design decision made here.

**Comment by alanhdu** [comment]
Closed by #5255
